cardano-cli/src/Cardano/CLI/EraBased/Options/Key.hs

```diff
@@ -21,65 +21,78 @@ import qualified Options.Applicative as Opt
 {- HLINT ignore "Use <$>" -}
 {- HLINT ignore "Move brackets to avoid $" -}
 
-pKeyCmds :: Parser (KeyCmds era)
+pKeyCmds :: Maybe (Parser (KeyCmds era))
 pKeyCmds =
-  asum
```

Comment:
Why the change from `Parser (..)` -> `Maybe (Parser (...))`

Reply:
The change represents a shift of responsibilities.

Previously it was the parent command group that decides whether the command should appear in an era.

The change shifts the responsibility from the parent to the command in question.  ie the command itself determines whether or not it should show up in a given era.

This is already how it is done for the governance commands and some other commands.

The purpose of this PR is to harmonise our code base to consistently use the latter convention.